Read me updated for AEPSampleNewArcEnabled

namArora3112 opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description

<!--- Describe your changes in detail -->

## Related Issue

<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->

## Motivation and Context

<!--- Why is this change required? What problem does it solve? -->

## How Has This Been Tested?

<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## Screenshots (if appropriate):

## Types of changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

cacheung commented on apps/AEPSampleAppNewArchEnabled/README.md:
```diff
@@ -1,50 +1,93 @@
-# Welcome to your Expo app 👋
+# Welcome to Your Expo AEPSample App with New Architecture Enabled 👍
```
Welcome to the Expo AEPSample App with New Architecture Enabled!

Let remove the emoji.

namArora3112 replied:
removed

cacheung commented on apps/AEPSampleAppNewArchEnabled/README.md:
````diff
@@ -1,50 +1,93 @@
+### Run Instructions for iOS
+1. Ensure you are in the root directory.
+   ```bash
+   yarn sampleappnewarchenabled:ios:run
````
did we do this already in the previous steps?  We just need one of them.

namArora3112 replied:
In earlier steps we have just installed dependencies for iOS but have not run the app

cacheung commented:
Btw, @namArora3112 Can you also make sure the existing AEPSampleAPP is still working in this repo?

namArora3112 commented on apps/AEPSampleAppNewArchEnabled/README.md:
```diff
@@ -1,50 +1,93 @@
+- **React Native**: [Tested with version 0.75.x](https://reactnative.dev/).
+- **Expo**: [Tested with version 51](https://docs.expo.dev/).
+- Note: The new architecture is enabled by default in this build.
 
```
I am currently building this app on ladybug (latest android studio) so can you suggest me if we should add it or not

cacheung replied:
Yes, let make the min Android sample app requirement to ladybug, so the users won't run that on the older version and encounter problem.

namArora3112 replied:
Added

namArora3112 commented on apps/AEPSampleAppNewArchEnabled/README.md:
```diff
@@ -1,50 +1,93 @@
 
-In the output, you'll find options to open the app in a
+## Installation
+To install **AEPSample App**, follow these steps:
```
corrected

namArora3112 commented on apps/AEPSampleAppNewArchEnabled/README.md:
````diff
@@ -1,50 +1,93 @@
-
-## Get a fresh project
+```
+yarn install && yarn sampleapp:ios:pod:install
````
fixed

namArora3112 commented on apps/AEPSampleAppNewArchEnabled/README.md:
````diff
@@ -1,50 +1,93 @@
-```bash
-npm run reset-project
 ```
+yarn run build
````
I have fixed it in this [PR](https://github.com/adobe/aepsdk-react-native/pull/403) . so can approve this for now

namArora3112 commented on apps/AEPSampleAppNewArchEnabled/README.md:
```diff
@@ -1,50 +1,93 @@
+### Configure the App ID
+
+- **iOS**: In `apps/AEPSampleAppNewArchEnabled/ios/AEPSampleAppNewArchEnabled/AdobeBridge.m`, find the `configureWithAppId` call and add your app ID.
+- **Android**: In `apps/AEPSampleApp/android/app/src/main/java/com/aepsampleapp/MainApplication.kt`, find the `configureWithAppId` call and add your app ID.
```
fixed

namArora3112 commented on apps/AEPSampleAppNewArchEnabled/README.md:
````diff
@@ -1,50 +1,93 @@
+
+2. Alternatively, run the iOS app from Xcode:
+   ```bash
+   npx expo start
````
fixed it

namArora3112 commented on apps/AEPSampleAppNewArchEnabled/README.md:
````diff
@@ -1,50 +1,93 @@
+### Run Instructions for Android
+
+1. Ensure an Android emulator is running or a device is connected. For guidance, refer to [Android Emulator Command Line](https://developer.android.com/studio/run/emulator-commandline).
+   ```bash
````
fixed

namArora3112 commented on apps/AEPSampleAppNewArchEnabled/README.md:
````diff
@@ -1,50 +1,93 @@
+   yarn sampleappnewarchenabled:android:run
+   ```
+
+2. Alternatively, run the app from Android Studio:
````
fixed

namArora3112 commented:
> Btw, @namArora3112 Can you also make sure the existing AEPSampleAPP is still working in this repo? This could be the reason why CICD is failing.

As discussed with you and @yangyansong-adbe ,  we have added CI/CD for our new aepSampleApp with new arch enabled and removed it for our earlier app . [
PR](https://github.com/adobe/aepsdk-react-native/pull/403)

cacheung commented on apps/AEPSampleAppNewArchEnabled/README.md:
````diff
@@ -1,50 +1,111 @@
-   npm install
-   ```
+## Prerequisites
+Ensure that your environment meets the following requirements before running this project:
````
running this sample app project:

(so we clarify that the environment requirement is for the sample app, not sdk)

namArora3112 replied:
fixed

cacheung commented on apps/AEPSampleAppNewArchEnabled/README.md:
```diff
@@ -1,50 +1,93 @@
+Ensure that your environment meets the following requirements before running this project:
 
-2. Start the app
+- Familiarity with [React Native development environment setup](https://reactnative.dev/docs/environment-setup) is recommended (preferably with the React Native CLI Quickstart).
```
Let remove "(preferably with the React Native CLI Quickstart)."

namArora3112 replied:
removed

cacheung reviewed (approved):
Few couple minor comments

Can we change "example" to "YOUR-APP-ID" to be consistent with iOS?
https://github.com/namArora3112/aepsdk-react-native/blob/read-me-update/apps/AEPSampleAppNewArchEnabled/android/app/src/main/java/com/AEPSampleAppNewArchEnabled/MainApplication.kt#L77

namArora3112 commented on apps/AEPSampleAppNewArchEnabled/README.md:
```diff
@@ -1,50 +1,113 @@
-# Welcome to your Expo app 👋
+# Welcome to Your Expo AEPSample App with New Architecture Enabled
```
fixed

namArora3112 commented:
> A couple minor comments
> 
> Can we change "example" to "YOUR-APP-ID" to be consistent with iOS? https://github.com/namArora3112/aepsdk-react-native/blob/read-me-update/apps/AEPSampleAppNewArchEnabled/android/app/src/main/java/com/AEPSampleAppNewArchEnabled/MainApplication.kt#L77

Have made it consistent with iOS
